src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -400,4 +400,85 @@ describe('ReactServerRendering', function() {
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
   });
+
+  it('warns with a no-op when an async setState is triggered', function() {
+    var Component = React.createClass({
+      componentWillMount: function() {
+        this.setState({text: 'hello'});
+        setTimeout(() => {
+          this.setState({text: 'error'});
+        });
+      },
+      render: function() {
+        return <div onClick={() => {}}>{this.state.text}</div>;
+      },
+    });
+
+    spyOn(console, 'error');
```

Comment:
@gaearon any particular reason to use `spyOn` instead of `jest.fn()`? I know @cpojer has recommended using `jest.fn` over `spyOn` to me in the past.